File: 0217-contains-duplicate/0217-contains-duplicate.ts
function containsDuplicate(nums: number[]): boolean {
    let hash = new Map<number, boolean>();

    for (let i = 0; i < nums.length; i++)   {
        if (hash.get(nums[i]))  {
            return true;
        }
        hash.set(nums[i], true)
    }
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.go
func containsDuplicate(nums []int) bool {
    hashMap := make(map[int]bool);

    for i := 0; i < len(nums); i++ {
        if hashMap[nums[i]]   {
            return true;
        }
        hashMap[nums[i]] = true; 
    }

    return false;
}